samples/splitter/splitter.cpp

```diff
@@ -521,6 +545,23 @@ void MySplitterWindow::OnPositionChanged(wxSplitterEvent& event)
     wxLogStatus(m_frame, "Position has changed, now = %d (or %d)",
                 event.GetSashPosition(), GetSashPosition());
 
+    // This event is only sent when the user manually dragged the sash.
+    // In this case we accept the user input so the sash is locked at the
+    // new position. If the sash is not locked, this has no effect but
+    // doesn't hurt either.
+    m_frame->SetSashPos(event.GetSashPosition());
+
+    event.Skip();
+}
+
+void MySplitterWindow::OnPositionResize(wxSplitterEvent &event)
+{
+    // When the splitter is resizing we only allow the sash to be moved
+    // if it is not locked. Otherwise we hold it at the position
+    // the user specified by manually dragging.
+    if (m_frame->IsSashLocked())
+        event.SetSashPosition(m_frame->GetSashPos());
+
     event.Skip();
```

Comment:
I'm surprised that calling `Skip()` still works, but if it does, it's actually a bug in `wxSplitterWindow`: if the event was skipped, it should _not_ be taken into account, i.e. there are 2 problems:

1. In `wxSplitterWindow` itself: it should ignore the position set by this event handler if it skips it.
2. In the sample: it shouldn't skip the event when handling it.

Reply:
The problem with Skip() is indeed a bug in `wxSplitterWindow`. The bug is with the `DoSendEvent` function. If the event is not vetoed, it is still allowed. So even if the handler was not found, it returns true because the event was not vetoed. I verified this in the sample with the `OnPositionChanging` handler which also uses Skip() and shows the same behavior. It was never noticed because the handler just prints the values, but doesn't change them.

When I added
```
event.SetSashPosition(100);
```
to it, it shows the same behavior as with resize, so this is an existing bug. I fix it now.

Reply:
Fixed bug with handling event.Skip() state and updated the samplecode to no longer call event.Skip().